src/dmd/parse.d

```diff
@@ -4770,6 +4770,8 @@ final class Parser(AST) : Lexer
                     error("'%s' cannot be placed after a template constraint", token.toChars);
                 else if (t == TOK.at)
                     error("attributes cannot be placed after a template constraint");
+                else if (t == TOK.if_)
+                    error("cannot use conditional definition for non-template functions. Use `static if` instead");
```

Comment:
I don't think "conditional definition" is the right terminology here.  Shouldn't it be "conditional expression"?

Reply:
I think that "function constraint" would be best.